### ISSUE TITLE
[color-slider] inside [color-picker] Jzazbz: jz and bz slider glitches

### ISSUE THREAD

**Issue by sidewayss** [color-js/elements#163]
In your docs page: https://elements.colorjs.io/src/color-picker/#dynamic

- select Jzazbz as the color space
- note that the top and bottom slider tracks are white
- slide the jz slider to the right and at one point you'll see the bz slider suddenly have color
- the thumbs on the sliders change color, but the track stays white.

**Comment by DmitrySharabin** [comment]
It looks like we have this issue because some of the slider's color stops have the value `lab(NaN% NaN NaN)`. This definitely should not happen.

**Comment by DmitrySharabin** [comment]
Actually, we have a FIXME comment from Lea there:
https://github.com/color-js/elements/blob/24ac3f7a67178f0b4cd5bc5da8b1bb3562e316fe/src/color-slider/color-slider.js#L74

@LeaVerou, could you please elaborate a bit on what should happen with such colors?

**Comment by sidewayss** [comment]
Is this simply a case of `null` needing to be converted to zero?  `null` used to be `NaN` in Color.js, and `parseFloat()` converts it to `NaN`. Excuse me if I'm way off base here.

**Comment by LeaVerou** [comment]
`null` should not usually be converted to `0`, that's a last ditch effort. If we're interpolating, `null` should be converted to the corresponding value in the previous/next stop (adding two stops if these are different).

**Comment by DmitrySharabin** [comment]
Digging into this a bit more, I learned that the actual line where we get value like `lab(NaN% NaN NaN)` is this one: https://github.com/color-js/elements/blob/2adcc7da35a99e33194e194cf6d7c8081099ca2c/src/color-slider/color-slider.js#L102

Precisely, the `display()` method. So, it's not that we work with colors with some coords equal to `null`, `none`, or `NaN`, but we have no displayable CSS colors for some color stops. What is the best strategy in this case? Is it possible to filter out colors we can't display and use the rest? Something like this:

```js
stops = stops.map(color => color.display()).filter(c => c.color.coords.every(coord => !Number.isNaN(coord))).join(", ");
```

However, we might end up with a thumb color that does not correspond to the slider's color at that point. And that's not correct, right?

<img width="546" alt="image" src="https://github.com/user-attachments/assets/51ed5c93-fc01-4232-bd36-019f60052159">

**Comment by sidewayss** [comment]
In this specific case, should there even be undisplayable stops that need filtering?  Or is there something wrong in specifying the undisplayable stops?  There should be a fully displayable set of stops for every supported color channel, right?

That's not to say you don't need to answer your general question of how to deal with illegitimate stops when they arise inside this element, if the user sets something wrong.  That might be more of an issue for `<color-slider>` than `<channel-slider>`.  I think I got the tag name wrong in the title for this issue.  This here is a `<channel-slider>`, right?

You just caused me to ask [this question](https://github.com/color-js/color.js/discussions/611) that I posted in the Discussions on the main color.js repo.

**Comment by DmitrySharabin** [comment]
> That might be more of an issue for `<color-slider>` than `<channel-slider>`. I think I got the tag name wrong in the title for this issue. This here is a `<channel-slider>`, right?

No-no, you tagged the element right. It's just easier for me to see the issue with the isolated channel. See
<img width="814" alt="image" src="https://github.com/user-attachments/assets/e9eca6a6-ebd0-4067-81e5-a446a15e5c76">

`<channel-slider>` uses `<color-slider>` under the hood.

**Comment by sidewayss** [comment]
> `<channel-slider>` uses `<color-slider>` under the hood.

Ha! I had it backwards based on reading the online docs.  Thanks for clarifying.

**Comment by DmitrySharabin** [comment]
Looks like the issue has been [fixed in Color.js](https://github.com/color-js/color.js/pull/635). I'm gonna close it. Feel free to reopen it if the issue is still present.
